Automattic/jetpack
# Subscriptions: Add Welcome Overlay behind the feature flag

Part of https://github.com/Automattic/jetpack/issues/36208

## Proposed changes:

This is the first iteration of the Welcome Overlay feature (p1HpG7-so7-p2).

It:
- opens the Welcome Overlay when a user opens the site home page or front page
- closes the overlay as soon as the subscription flow starts
- sets the cookie to prevent the overlay from opening on every page view (expires in 7 days)
- makes the overlay template editable

<img width="958" alt="Screenshot 2024-05-16 at 16 32 50" src="https://github.com/Automattic/jetpack/assets/4068554/a9a9885e-8b01-405e-a304-76de20719b78">

The feature is behind the feature flag and if you want to enable it you need to go into your sandbox mu-plugins, and add the following line to your `0-sandbox.php` file:
```php
add_filter( 'jetpack_subscribe_overlay_enabled', '__return_true' );
```

#### Known issues

There is no way of changing the overlay background color. I'll try to fix in in a follow-up PR.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






1. Test it without the feature flag set and make sure the overlay does not open and there is no suspicious warnings/notices in logs.
2. Enable the feature flag, visit the site home or front page making sure the overlay works as expected.
3. Go to `/wp-admin/site-editor.php?postType=wp_template_part&postId=visualblocks%2F%2Fjetpack-subscribe-overlay&canvas=edit`, edit the overlay content and make sure the changes are visible on the frontend (you'll need to delete the `jetpack_post_subscribe_overlay_dismissed` cookie to render the overlay again)
